pkg/controller/syncmachineset/gcpactuator_test.go

```diff
@@ -152,11 +139,9 @@ func TestGCPAcuator(t *testing.T) {
 				}
 				return cd
 			}(),
-			mockGCPClient: func(mockCtrl *gomock.Controller) *mockgcp.MockClient {
-				client := mockgcp.NewMockClient(mockCtrl)
-				mockListComputeImage(client, []string{"imageA"})
-				mockListComputeZones(client, []string{"zoneA"})
-				return client
+			mockGCPClient: func(client *mockgcp.MockClient, cd *hivev1.ClusterDeployment) {
+				mockListComputeImage(client, []string{"imageA"}, cd.Status.InfraID)
+				mockListComputeZones(client, []string{"zoneA"}, "")
```

Comment:
just trying to get the code far enough so that when we call into the installer's MachineSet() generation we can simulate a failure

Reply:
OK. And that is entirely separate from listing the images and listing the zones. So there is no need to add the separate wrinkle of an empty region.

Reply:
the empty region was because previously i had a nil spec.platform.gcp, but i can get the failure by filling out both spec.platform.aws and gcp simultaneously (even though that isn't really valid it still does the trick).

Reply:
I don't understand this. The region passed to `ListComputeZones` does not come from the clusterdeployment. Why does the contents of spec.platform factor in at all?

Reply:
region does come from the clusterDeployment now that we've remove region from the struct.